src/event_pusher/mod_event_pusher_push_plugin_defaults.erl

```diff
@@ -81,12 +89,13 @@ publish_notification(Acc0, From, #jid{lserver = Host} = To, Packet, Services) ->
 
 -spec push_notification_iq(From :: jid:jid(),
                            Packet :: exml:element(), Node :: mod_event_pusher_push:pubsub_node(),
-                           Form :: mod_event_pusher_push:form()) -> jlib:iq().
-push_notification_iq(From, Packet, Node, Form) ->
+                           Form :: mod_event_pusher_push:form(),
+                           MessageCount :: binary()) -> jlib:iq().
+push_notification_iq(From, Packet, Node, Form, MessageCount) ->
     ContentFields =
         [
          {<<"FORM_TYPE">>, ?PUSH_FORM_TYPE},
-         {<<"message-count">>, <<"1">>},
+         {<<"message-count">>, MessageCount},
```

Comment:
This is the point where we serialise our data into a on-the-wire format, so `integer_to_binary(MessageCount)`. See the comment to line 61.